email/resource_email.go

```diff
@@ -83,21 +99,51 @@ func resourceEmailCreate(d *schema.ResourceData, m interface{}) error {
 		preamble + "\n\n" +
 		body
 
-	err := smtp.SendMail(smtpServer+":"+smtpPort,
-		smtp.PlainAuth("", smtpUsername, smtpPassword, smtpServer),
-		from, []string{to}, []byte(msg))
-
+	// TODO: make this tf configurable
+	maxRetries := 5
+	// send mail using exponential back-off
+	err := sendMail(smtp.SendMail, maxRetries, smtpServer, smtpPort, smtpUsername, smtpPassword, from, to, msg)
+	// log error if not cleared after retries
 	if err != nil {
 		log.Printf("smtp error: %s", err)
 		return err
 	}
 
-	// Create unique ID using current timestamp
 	timestamp := time.Now().Unix()
 	d.SetId(to + " | " + subject + " | " + strconv.FormatInt(timestamp, 10))
+
 	return resourceEmailRead(d, m)
 }
 
+func sendMail(sendEmail SendMailFunc, maxRetries int, smtpServer string, smtpPort string, smtpUsername string, smtpPassword string, from string, to string, msg string) error {
```

Comment:
```suggestion
func sendMail(sendMailImpl SendMailFunc, maxRetries int, smtpServer string, smtpPort string, smtpUsername string, smtpPassword string, from string, to string, msg string) error {
```

Just so that people don't confuse `sendMail` and `sendEmail`